sky/task.py

```diff
@@ -212,6 +212,7 @@ def __init__(
         blocked_resources: Optional[Iterable['resources_lib.Resources']] = None,
         # Internal use only.
         file_mounts_mapping: Optional[Dict[str, str]] = None,
+        metadata: Optional[Dict[str, Any]] = None,
```

Comment:
If we want to add this field, we also need to add it to `to_yaml_config` and `from_yaml_config`, otherwise it may get dropped if we serialize and deserialize the yaml. We do this even for internal fields like `file_mounts_mapping`.

Reply:
I kind of feel that it may be simpler to avoid adding it to the task and just get the hash during _execute in cloud_vm_ray_backend.py, and pass it into add_job/set_pending.

Reply:
I put it in `Task` bc it feels like the git commit hash is a property of that task like this task is on commit 1234 and in case we want to add other metadata to tasks.